### ISSUE TITLE
Ambiguous documentation for `append` argument to `MultiPattern::reparse`

### ISSUE THREAD

**Issue by alexrutar** [helix-editor/nucleo#66]
The docs for [`MultiPattern::reparse`](https://docs.rs/nucleo/latest/nucleo/pattern/struct.MultiPattern.html#method.reparse) explain that the 'append' argument should be set to `true` if the new query string contains the old query string as a prefix.

However, it seems to me that this does consider the case when there is a trailing `\`? I had to add this extra check downstream in `nucleo_picker` in order to get proper search results.

It seems plausible to me that this is also a bug in `helix`: https://github.com/helix-editor/helix/blob/db1d84256fbae21abb3ba46943fb1abb8e211355/helix-term/src/ui/picker.rs#L539

I guess either this is actually an error (and the reparse method should be fixed) or it should be clarified in the documentation that one also needs to check for the presence of a trailing `\`.

**Comment by pascalkuthe** [comment]
Yeah that is an oversight, should just include a check wether the pattern ends with a backslash. Altough I vaguely remeber I did include that but maybe forgot somewhere or got lost
